packages/database/marsh/marshalling_dynamo_repository.go

```diff
@@ -0,0 +1,55 @@
+package marsh
+
+import (
+	"reflect"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/juju/errors"
+	"theodo.red/creditcompanion/packages/database"
+	"theodo.red/creditcompanion/packages/database/tdynamo"
+)
+
+type MarshallingDynamoRepository struct {
+	db        tdynamo.DynamoDbInterface
+	tableName string
+}
+
+func (r *MarshallingDynamoRepository) Get(id string, dest interface{}) error {
+	if reflect.TypeOf(dest).Kind() != reflect.Ptr {
+		return errors.New("Bad destination pointer passed to marshalling repository.")
+	}
+	preMarshalReplica := reflect.ValueOf(dest).Elem().Interface()
+
+	result, getErr := r.db.GetItem(&dynamodb.GetItemInput{
+		TableName: aws.String(r.tableName),
+		Key: map[string]*dynamodb.AttributeValue{
+			"Id": {
+				S: &id,
+			},
+		}})
+	if getErr != nil {
+		// Don't log errors til you handle them
+		return errors.Annotate(getErr, "Call to retrieve "+id+" from db failed.")
+	}
+
+	if result.Item == nil {
+		return errors.New("Item with id " + id + " could not be found.")
+	}
+
+	unmarshalErr := dynamodbattribute.UnmarshalMap(result.Item, dest)
```

Comment:
If needs be, later, we can add a marshaller to take the passed in type (whatever it may be) and having dynamo populate it.
Dynamo will do a best effort to match its fields to the provided struct itself, ignoring things such as casing if needs be. This means that right now that our struct types that we pass into here, their fields will map directly to their serialised database form. This is not a problem unless we add complex fields that require (de/serialistation), but for now is acceptable.